medmodels/treatment_effect_estimation/treatment_effect.py

```diff
@@ -0,0 +1,607 @@
+        follow_up_period_days: int = 365,
+        follow_up_period_reference: Literal["first", "last"] = "last",
+        outcome_before_treatment_days: Optional[int] = None,
+        filter_controls_operation: Optional[NodeOperation] = None,
+        matching_method: Optional[MatchingMethod] = None,
+        matching_essential_covariates: MedRecordAttributeInputList = ["gender", "age"],
+        matching_one_hot_covariates: MedRecordAttributeInputList = ["gender"],
+        matching_model: Model = "logit",
+        matching_distance_metric: Metric = "absolute",
+        matching_number_of_neighbors: int = 1,
+        matching_hyperparam: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        Initializes a Treatment Effect analysis setup with specified treatments and
+        outcomes within a medical record dataset.
+        Validates the presence of specified dimensions and attributes within the
+        provided MedRecord object, ensuring the specified treatments and outcomes are
+        valid and available for analysis.
+
+        Args:
+            treatment (Group): The group of treatments to analyze.
+            outcome (Group): The group of outcomes to analyze.
+            patients_group (Group, optional): The group of patients to analyze.
+                Defaults to "patients".
+            time_attribute (MedRecordAttribute, optional): The time attribute to use for
+                time-based analysis. Defaults to "time".
+            washout_period_days (Dict[str, int], optional): The washout period in days
+                for each treatment group. Defaults to dict().
+            washout_period_reference (Literal["first", "last"], optional): The reference
+                point for the washout period. Defaults to "first".
+            grace_period_days (int, optional): The grace period in days after the
+                treatment. Defaults to 0.
+            grace_period_reference (Literal["first", "last"], optional): The reference
+                point for the grace period. Defaults to "last".
+            follow_up_period_days (int, optional): The follow-up period in days after
+                the treatment. Defaults to 365.
+            follow_up_period_reference (Literal["first", "last"], optional): The
+                reference point for the follow-up period. Defaults to "last".
+            outcome_before_treatment_days (Optional[int], optional): The number of days
+                before the treatment to consider for outcomes. Defaults to None.
+            filter_controls_operation (Optional[NodeOperation], optional): An optional
+                operation to filter the control group based on specified criteria.
+                Defaults to None.
+            matching_method (Optional[MatchingMethod]): The method to match treatment
+                and control groups. Defaults to None.
+            matching_essential_covariates (Optional[List[MedRecordAttribute]], optional):
```

Comment:
Not Optional[List[MedRecordAttribute]] but: MedRecordAttributeInputList, optional. Also the description fails to indicate the default value.